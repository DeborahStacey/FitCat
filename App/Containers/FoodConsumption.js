import React from 'react'
import { AsyncStorage, Picker, Text, TextInput, TouchableOpacity, ScrollView, View } from 'react-native'
import { default as StorageKeys } from '../Config/StorageKeys'
import Icon from 'react-native-vector-icons/FontAwesome'

import styles from './Styles/FoodConsumptionStyle'

/*
Name
Brand OPTIONAL
Description OPTIONAL
*/

export default class FoodConsumption extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      foodTypes: [],
      hideFoodForm: true,
      formNameText: '',
      formBrandText: '',
      formDescText: '',
      selectedFood: ''
    }

    this.imlazydothing()
  }

  imlazydothing = async () => {
    this.setState({foodTypes: await this.getFoodTypes()})
  }

  handleAddFoodTypePress = () => {
    this.setState({
      hideFoodForm: false
    })
  }

  resetAsyncStorage = async () => {
    try {
      await AsyncStorage.setItem(StorageKeys.SAVED_FOOD_TYPES, '')
      this.setState({foodTypes: []})
    } catch (error) {
      console.log(error)
    }
  }

  getFoodTypes = async () => {
    var foodTypes = []
    try {
      var stringFoodTypes = await AsyncStorage.getItem(StorageKeys.SAVED_FOOD_TYPES)
      if (stringFoodTypes != null) {
        foodTypes = JSON.parse(stringFoodTypes)
      }
    } catch (error) {
      console.log('Failed trying to read food types from AsyncStorage')
      console.log(error)
      foodTypes = []
    }

    console.log(foodTypes)
    return foodTypes
  }

  saveFoodTypes = async () => {
    var newFoodType = {
      name: this.state.formNameText,
      brand: this.state.formBrandText,
      description: this.state.formDescText
    }

    var foodTypes = this.state.foodTypes
    foodTypes.push(newFoodType)
    try {
      await AsyncStorage.setItem(StorageKeys.SAVED_FOOD_TYPES, JSON.stringify(foodTypes))
    } catch (error) {
      console.log('Failed trying to save food types to AsyncStorage')
      console.log(error)
    }

    console.log('SAVING FOOD TYPES')
    console.log(foodTypes)
    this.setState({foodTypes: foodTypes})
  }

  buildFoodTypeDisplay = () => {
    var foodTypes = this.state.foodTypes
    var displayedFoodTypes = []
    if (foodTypes === null || foodTypes === undefined) {
      foodTypes = []
    }
    console.log(foodTypes)
    foodTypes.forEach((foodType, i) => {
      displayedFoodTypes.push(<Picker.Item key={i} style={styles.sectionText} label={foodType.name} value={foodType.name} />)
    })
    return displayedFoodTypes
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <ScrollView style={styles.container}>
          <TouchableOpacity style={styles.addFoodTypeContainer} onPress={this.handleAddFoodTypePress}>
            <Text style={styles.sectionText}>Add a new food type</Text>
            <Icon style={styles.addFoodTypeIcon} name='plus' size={20} />
          </TouchableOpacity>
          {this.state.hideFoodForm ? null
            : <View style={styles.container}>
              <TextInput
                style={{height: 40}}
                placeholder='Name (Required)'
                onChangeText={(text) => this.setState({
                  formNameText: text
                })}
              />
              <TextInput
                style={{height: 40}}
                placeholder='Brand'
                onChangeText={(text) => this.setState({
                  formBrandText: text
                })}
              />
              <TextInput
                style={{height: 40}}
                placeholder='Description'
                onChangeText={(text) => this.setState({
                  formDescText: text
                })}
              />
              <TouchableOpacity onPress={this.saveFoodTypes}>
                <Text style={styles.sectionText}>Click</Text>
              </TouchableOpacity>
              <TouchableOpacity onPress={this.resetAsyncStorage}>
                <Text style={styles.sectionText}>Clear</Text>
              </TouchableOpacity>
            </View>
          }
          {
            this.state.foodTypes && this.buildFoodTypeDisplay()
          }
        </ScrollView>
      </View>
    )
  }
}
