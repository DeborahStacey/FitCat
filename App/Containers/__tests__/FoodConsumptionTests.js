import 'react-native'
import React, { Component } from 'react'
import FoodConsumption from '../FoodConsumption'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('Picker', () => {
  const React = require('React');
  return class MockPicker extends Component {
    static Item = props => React.createElement('Item', props, props.children);
    static propTypes = { children: React.PropTypes.any };

    render() {
      return React.createElement('Picker', this.props, this.props.children);
    }
  }
})
it('renders the food consumption page correctly', () => {
  const tree = renderer.create(
    <FoodConsumption />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
