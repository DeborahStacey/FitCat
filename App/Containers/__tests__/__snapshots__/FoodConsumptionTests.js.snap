exports[`test renders the food consumption page correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        Add a new food type
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "color": "#2C457F",
              "marginLeft": 10,
            },
          ]
        }>
        ÔÅß
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 25,
        }
      }>
      <Picker
        onValueChange={[Function]}
        selectedValue={-1}
        style={
          Object {
            "borderWidth": 1,
          }
        }>
        <Item
          label="[missing \"en.no_saved_food\" translation]"
          value={Object {}} />
      </Picker>
    </View>
    <View
      style={
        Object {
          "padding": 25,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.cups_food_prompt\" translation]
      </Text>
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Cups"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "paddingLeft": 10,
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#466ECC",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 5,
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        [MISSING \"EN.SUBMIT_FOOD_RECORD\" TRANSLATION]
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#466ECC",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 5,
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        [MISSING \"EN.VIEW_FOOD_CONSUMPTION\" TRANSLATION]
      </Text>
    </View>
  </ScrollView>
</View>
`;
